executor.lua

```diff
@@ -31,9 +31,18 @@ missions.start = function(pos, player)
 		description = meta:get_string("description") or ""
 	}
 
+	local nointeract = meta:get_int("nointeract")
+
+	if nointeract then
+		minetest.set_player_privs(playername, {
+			interact = false,
+			missions_nointeract = true })
```

Comment:
This will grant missions_nointeract but revoke all privs other than interact. Maybe start from the values returned by `get_player_privs`?